### PR TITLE
move `verify-blob` function to "pkg/cosign" to be consistent with `verify`

### PR DESCRIPTION
**Pull request by hirokuni-kitahara**
Signed-off-by: Hirokuni-Kitahara1 <hirokuni.kitahara1@ibm.com>

#### Summary

This PR solves the issue https://github.com/sigstore/cosign/issues/2222 by moving the core function of `verify-blob` from "cmd/cosign/cli/verify" package to "pkg/cosign" package.
The new function `VerifyBlobSignature()` is based on basically the same design as `VerifyImageSignature()`, which is the core function of `verify` command.
Also the arguments of `VerifyBlobSignature()` are all non-filepath information such as []byte, so it is easy for developers to invoke this function even on some read-only environment.

### CODE REVIEW

File: pkg/cosign/verify_blob.go [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+			if isIntotoDSSE(blobBytes) {
+				verifier = dsse.WrapVerifier(verifier)
+			}
+			verifiers = append(verifiers, verifier)
+		}
+	}
+
+	var validSigExists bool
+	for _, verifier := range verifiers {
+		if err := verifyOCISignature(ctx, verifier, sig); err != nil {
```
Comment (asraa) [inline pkg/cosign/verify_blob.go:129]: What happens to intoto attestations?

File: pkg/cosign/verify_blob.go [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+We recommend requesting the certificate/signature from the original signer of this blob and manually verifying with cosign verify-blob --cert [cert] --signature [signature].`)
+		return checkedSignatures, bundleVerified, fmt.Errorf("could not find a valid tlog entry for provided blob, found %d invalid entries", len(verifiers))
+	}
+
+	if validSigExists {
+		fmt.Fprintln(os.Stderr, "Verified OK")
+		checkedSignatures = append(checkedSignatures, sig)
+	}
+
+	if !validSigExists && co.RekorClient != nil {
```
Comment (asraa) [inline pkg/cosign/verify_blob.go:148]: How come the TLOG entry time will only be checked if a valid sig does not exist? `verifyOCISignature` does not check cert expiry time.

**Review by asraa — COMMENTED** [review]
Hi! Do you think we could hold off on this for a day or two?

**Comment by haydentherapper** [comment]
We should hold off on moving this code until a) the refactor we are working on is done b) there are most tests so we have confidence in moving code.

**Comment by haydentherapper** [comment]
Another meta comment, seems like there are functional changes mixed into this. A refactor should be as minimal as possible, so if we did want to move this, it should be clear that we are just moving the functions, not making any changes in them.

File: pkg/cosign/verify_blob.go (hunk 1 above)
Comment (hirokuni-kitahara) [reply pkg/cosign/verify_blob.go:129]: Thank you for your reviewing! The intoto DSSE verifier will only be available in keyless verification mode with my current codes, and this is simply a mistake.
I will fix this.

File: pkg/cosign/verify_blob.go (hunk 2 above)
Comment (hirokuni-kitahara) [reply pkg/cosign/verify_blob.go:148]: Thank you for the information, I think we should have `VerifyBundle()` after `verifyOCISignature()` as VerifyImageSignature does. I will fix this.

**Comment by hirokuni-kitahara** [comment]
@asraa Thank you for your reviewing and your comments, I will update the changes to fix some problems you mentioned in the comments.

@haydentherapper Thank you for your comment, I totally agree with that we should not mix a refactor and functional changes in 1 PR. I will remove functional changes from this PR and will create another PR for it when the timing is good.
About the refactoring you are working on, could you share some timeline like when the verify-blob codes would be ready for moving if possible?

**Comment by haydentherapper** [comment]
We'll hopefully have the refactor in by either end of this week or next week.

**Comment by hirokuni-kitahara** [comment]
That's great! Thank you very much @haydentherapper

**Pull request closed without merging**